types/src/validator_signer.rs

```diff
@@ -82,20 +79,24 @@ pub mod proptests {
     use proptest::{prelude::*, sample, strategy::LazyJust};
 
     #[allow(clippy::redundant_closure)]
-    pub fn arb_signing_key() -> impl Strategy<Value = Ed25519PrivateKey> {
+    pub fn arb_signing_key() -> impl Strategy<Value = bls12381::PrivateKey> {
         prop_oneof![
             // The no_shrink here reflects that particular keypair choices out
             // of random options are irrelevant.
-            LazyJust::new(|| Ed25519PrivateKey::generate_for_testing()),
-            LazyJust::new(|| Ed25519PrivateKey::genesis()),
+            LazyJust::new(|| bls12381::PrivateKey::generate_for_testing()),
+            LazyJust::new(|| bls12381::PrivateKey::genesis()),
         ]
     }
 
     pub fn signer_strategy(
-        signing_key_strategy: impl Strategy<Value = Ed25519PrivateKey>,
+        signing_key_strategy: impl Strategy<Value = bls12381::PrivateKey>,
     ) -> impl Strategy<Value = ValidatorSigner> {
-        signing_key_strategy
-            .prop_map(|signing_key| ValidatorSigner::new(AccountAddress::random(), signing_key))
+        signing_key_strategy.prop_map(|signing_key| {
+            ValidatorSigner::new(
+                AccountAddress::from_bytes(&signing_key.public_key().to_bytes()[0..32]).unwrap(),
```

Comment:
1. it's a fix to the previous code, the address has to be derived from the key
2. well someone blow it up will need to fix it since it's ll just panic the test :)

Reply:
Not very good to have arbitrary constants like that throughout the code though...

Reply:
agreed, we should support a derive function similar to ed25519 like https://github.com/aptos-labs/aptos-core/blob/09c8b36e7461ff6efa802ad13c7c83f266311c49/types/src/account_address.rs#L13

Reply:
Is it hard to add that in now?